python/ray/autoscaler/_private/util.py

```diff
@@ -98,9 +98,28 @@ def fillout_defaults(config: Dict[str, Any]) -> Dict[str, Any]:
     defaults = _get_default_config(config["provider"])
     defaults.update(config)
     defaults["auth"] = defaults.get("auth", {})
+    try:
+        defaults = _fillout_available_node_types_resources(defaults)
+    except Exception:
+        # We don't want to introduce new errors with filling available node
+        # types resources feature.
+        pass
```

Comment:
Please don't ever swallow exceptions--- instead logger.exception("Failed to fill out available node types") here.

Reply:
I added the logger initially but I was reluctant to keep it as it seemed to me like I am adding a feature that should help the user and make his life easier but it failed to run so why would I tell the user that it failed (It seems like the feature introduced the problem)? But based on your comment I will keep the logger.exception.